core/common/src/Utf8.kt

```diff
@@ -196,24 +205,29 @@ public fun Source.readString(byteCount: Long): String {
 }
 
 /**
- * Removes and returns a single UTF-8 code point, reading between 1 and 4 bytes as necessary.
+ * Decodes a single code point value from UTF-8 code units, reading between 1 and 4 bytes as necessary.
  *
  * If this source is exhausted before a complete code point can be read, this throws an
  * [EOFException] and consumes no input.
  *
- * If this source doesn't start with a properly-encoded UTF-8 code point, this method will remove
+ * If this source doesn't start with a properly encoded UTF-8 code point, this method will remove
  * 1 or more non-UTF-8 bytes and return the replacement character (`U+fffd`). This covers encoding
- * problems (the input is not properly-encoded UTF-8), characters out of range (beyond the
+ * problems (the input is not properly encoded UTF-8), characters out of range (beyond the
```

Comment:
> If this source doesn't start with a properly encoded UTF-8 code point, this method will remove
1 or more non-UTF-8 bytes and return the replacement character (`U+fffd`). This covers <...> code points for UTF-16 surrogates (`U+d800`..`U+dfff`)

What this means, exactly? We can decode 1 whole code point that later forms a surrogate pair. Then in what case UTF-16 surrogates are replaced with a replacement char?

Reply:
> What this means, exactly?

If a prefix of data contained in a `Source` is an ill-formed code unit sequence from the UTF-8 point of view, this method will remove these code units from the source and return the replacement code point.

If there's a well-formed code unit sequence, it will be removed from the Source and decoded. If a decoded value falls out of valid UTF-16 code point value ranges, we'll also return the replacement code point.

I think that rephrasing `If this source doesn't start with a properly encoded UTF-8 code point` to `If this source starts with an ill-formed UTF-8 code unit sequence` should help with translating the idea. WDYT?